tests/api/runtimes/test_kubejob.py

```diff
@@ -141,6 +147,275 @@ def test_run_with_node_selection(self, db: Session, client: TestClient):
             expected_affinity=affinity,
         )
 
+    def test_change_preemptible_modes_between_runs(
+        self, db: Session, client: TestClient
+    ):
+        node_selector = self._generate_node_selector()
+        mlrun.mlconf.preemptible_nodes.node_selector = base64.b64encode(
+            json.dumps(node_selector).encode("utf-8")
+        )
+        runtime = self._generate_runtime()
+        runtime.with_preemption_mode(PreemptionModes.prevent)
+        self._execute_run(runtime)
+        self.assert_node_selection(affinity=self._generate_preemptible_anti_affinity())
+
+        runtime.with_preemption_mode(PreemptionModes.constrain)
+        self._execute_run(runtime)
+        self.assert_node_selection(affinity=self._generate_preemptible_affinity())
+
+        runtime.with_preemption_mode(PreemptionModes.allow)
+        self._execute_run(runtime)
+        self.assert_node_selection()
+
+        runtime.with_preemption_mode(PreemptionModes.constrain)
+        self._execute_run(runtime)
+        self.assert_node_selection(affinity=self._generate_preemptible_affinity())
+
+        runtime.with_preemption_mode(PreemptionModes.prevent)
+        self._execute_run(runtime)
+        self.assert_node_selection(affinity=self._generate_preemptible_anti_affinity())
+
+        # set default preemptible tolerations
+        tolerations = self._generate_tolerations()
+        serialized_tolerations = self.k8s_api.sanitize_for_serialization(tolerations)
+        mlrun.mlconf.preemptible_nodes.tolerations = base64.b64encode(
+            json.dumps(serialized_tolerations).encode("utf-8")
+        )
+        runtime.with_preemption_mode(PreemptionModes.constrain)
+        self._execute_run(runtime)
+        self.assert_node_selection(
+            affinity=self._generate_preemptible_affinity(), tolerations=tolerations
+        )
+
+        runtime.with_preemption_mode(PreemptionModes.prevent)
+        self._execute_run(runtime)
+        self.assert_node_selection(
+            affinity=self._generate_preemptible_anti_affinity(),
+        )
+
+    def test_run_with_prevent_preemptible_mode(self, db: Session, client: TestClient):
```

Comment:
This test should test both the merging and pruning logic:
* merging - having your own tolerations set, making sure the configured tolerations are being added on top
* pruning - having your own tolerations set, making sure only the configured tolerations are being removed
I would also do the same two tests when tolerations for preemptible nodes are not configured